[Docs] Add docs for LoyaltyRule[Api] (#12289)
Sylius/Sylius: docs/book/loyalty/loyalty_rule.rst

.. rst-class:: plus-doc

Loyalty Rule
==============

Adding new Loyalty Rule Configuration to API
--------------------------------------------

After created a new loyalty rule configuration you need to add a new ``LoyaltyRuleActionDataTransformer`` that implements ``Sylius\Plus\Loyalty\Infrastructure\DataTransformer\LoyaltyRuleActionDataTransformerInterface``

> To be more flexible, creating new ``LoyaltyRuleConfiguration`` is created by new DTO object: ``Sylius\Plus\Loyalty\Application\DTO\LoyaltyRuleAction`` that has ``configuration`` field as an ``array``

Reviewer: ```suggestion
To be more flexible, new ``LoyaltyRuleConfiguration`` is created by new DTO object: ``Sylius\Plus\Loyalty\Application\DTO\LoyaltyRuleAction`` that has ``configuration`` field as an ``array``
```

> Array allow you to add any configuration and in the aftermath you have to transform it object to ``Sylius\Plus\Loyalty\Domain\Model\LoyaltyRuleConfiguration\LoyaltyRuleConfigurationInterface`` instance by your custom  ``LoyaltyRuleActionDataTransformer``

Reviewer: ```suggestion
That field allows you to add any configuration and in the aftermath, you have to transform this object to ``Sylius\Plus\Loyalty\Domain\Model\LoyaltyRuleConfiguration\LoyaltyRuleConfigurationInterface`` instance by your custom  ``LoyaltyRuleActionDataTransformer``
```
